dev/Breadcrumb/Breadcrumb.cpp

```diff
@@ -0,0 +1,561 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include "pch.h"
+#include "common.h"
+#include "Vector.h"
+#include "Breadcrumb.h"
+#include "RuntimeProfiler.h"
+#include "ResourceAccessor.h"
+
+#include "BreadcrumbItem.h"
+#include "BreadcrumbLayout.h"
+#include "BreadcrumbItemClickedEventArgs.h"
+
+Breadcrumb::Breadcrumb()
+{
+    __RP_Marker_ClassById(RuntimeProfiler::ProfId_Breadcrumb);
+
+    SetDefaultStyleKey(this);
+    m_breadcrumbElementFactory = winrt::make_self<BreadcrumbElementFactory>();
+}
+
+void Breadcrumb::OnApplyTemplate()
+{
+    winrt::IControlProtected controlProtected{ *this };
+
+    // TODO: Implement
+    m_breadcrumbItemRepeater.set(GetTemplateChildT<winrt::ItemsRepeater>(L"BreadcrumbItemRepeater", controlProtected));
+
+    if (auto const& thisAsIUIElement7 = this->try_as<winrt::IUIElement7>())
+    {
+        thisAsIUIElement7.PreviewKeyDown({ this, &Breadcrumb::OnChildPreviewKeyDown });
+    }
+
+    AccessKeyInvoked({ this, &Breadcrumb::OnAccessKeyInvoked });
+    GettingFocus({ this, &Breadcrumb::OnGettingFocus });
+
+    if (const auto& breadcrumbItemsRepeater = m_breadcrumbItemRepeater.get())
+    {
+        breadcrumbItemsRepeater.ItemsSource(winrt::make<Vector<IInspectable>>());
+        breadcrumbItemsRepeater.ItemTemplate(*m_breadcrumbElementFactory);
+        
+        m_itemRepeaterElementPreparedRevoker = breadcrumbItemsRepeater.ElementPrepared(winrt::auto_revoke, { this, &Breadcrumb::OnElementPreparedEvent });
+        m_itemRepeaterElementIndexChangedRevoker = breadcrumbItemsRepeater.ElementIndexChanged(winrt::auto_revoke, { this, &Breadcrumb::OnElementIndexChangedEvent });
+        m_itemRepeaterElementClearingRevoker = breadcrumbItemsRepeater.ElementClearing(winrt::auto_revoke, { this, &Breadcrumb::OnElementClearingEvent });
+
+        m_breadcrumbItemRepeaterLoadedRevoker = breadcrumbItemsRepeater.Loaded(winrt::auto_revoke, { this, &Breadcrumb::OnBreadcrumbItemRepeaterLoaded });
+    }
+
+    UpdateItemsSource();
+}
+
+void Breadcrumb::OnPropertyChanged(const winrt::DependencyPropertyChangedEventArgs& args)
+{
+    const winrt::IDependencyProperty& property = args.Property();
+
+    // TODO: Implement
+    if (property == s_ItemsSourceProperty)
+    {
+        UpdateItemsSource();
```

Comment:
I would recommend calling this "OnItemsSourceUpdated)"

Same for OnItemTemplateUpdated and OnDropdownItemTemplateUpdated below.